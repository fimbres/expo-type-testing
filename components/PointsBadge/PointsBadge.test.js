import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react-native';

import { PointsBadge } from '.';

describe('<PointsBadge />', () => {
    it('render text correctly', () => {
        render(<PointsBadge points={1000} />);
        const text = screen.getByText('1,000.00 pts');
        expect(text).toBeDefined();
    });

    it('render correctly', () => {
        const tree = renderer
        .create(
            <PointsBadge points={1000} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
