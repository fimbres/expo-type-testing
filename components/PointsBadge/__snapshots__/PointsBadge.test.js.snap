// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#334FFA",
      "borderRadius": 20,
      "elevation": 3,
      "marginHorizontal": 32,
      "paddingBottom": 49,
      "paddingHorizontal": 18,
      "paddingTop": 21,
      "shadowColor": "#020202",
      "shadowOffset": Object {
        "height": 3,
        "width": 2,
      },
      "shadowOpacity": 0.5,
      "shadowRadius": 8,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontFamily": "avenir-black",
        "fontSize": 16,
        "fontWeight": "800",
        "lineHeight": 21.86,
        "marginBottom": 7,
      }
    }
  >
    Diciembre
  </Text>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontFamily": "avenir-black",
        "fontSize": 32,
        "fontWeight": "800",
        "lineHeight": 43.71,
        "textAlign": "center",
      }
    }
  >
    1,000.00
     pts
  </Text>
</View>
`;
